// devexpress-dashboard/model/items/metadata/_text-box-item.js
/**
* DevExpress Dashboard (_text-box-item.js)
* Version:  19.2.12
* Build date: Apr 23, 2021
* Copyright (c) 2012 - 2021 Developer Express Inc. ALL RIGHTS RESERVED
* License: https://www.devexpress.com/Support/EULAs/universal.xml
*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _base_metadata_1 = require("../../metadata/_base-metadata");
var _data_dashboard_item_1 = require("./_data-dashboard-item");
var interactivity_options_1 = require("../options/interactivity-options");
var ko = require("knockout");
exports.textBoxText = {
    propertyName: "text",
    modelName: "Text",
    displayName: "DashboardWebStringId.TextBox.Text",
    defaultVal: "",
    editor: _base_metadata_1.editorTemplates.textFile,
    from: function (value) { return ko.observable(value); },
    toJsonObject: function (value) {
        return value.replace(/\0*$/g, '').trim();
    },
};
exports.textBoxValues = { propertyName: _base_metadata_1.valuesPropertyName, modelName: "Values", displayName: "DashboardStringId.DescriptionValues", array: true, editor: _base_metadata_1.editorTemplates.commonCollection, visible: false };
exports.textBoxDashboardItemSerializationsInfo = _data_dashboard_item_1.dataDashboardItemSerializationsInfo.concat([exports.textBoxValues, exports.textBoxText, interactivity_options_1._baseInteractivityOptionsMeta]);
