// devexpress-dashboard/designer/parameter-settings/_lookup-values-select-box-options.js
/**
* DevExpress Dashboard (_lookup-values-select-box-options.js)
* Version:  19.2.12
* Build date: Apr 23, 2021
* Copyright (c) 2012 - 2021 Developer Express Inc. ALL RIGHTS RESERVED
* License: https://www.devexpress.com/Support/EULAs/universal.xml
*/
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var dx_analytics_core_1 = require("@devexpress/analytics-core/dx-analytics-core");
var look_up_value_1 = require("../../model/parameters/look-up-value");
var data_source_1 = require("devextreme/data/data_source");
var _default_1 = require("../../data/localization/_default");
var model_1 = require("../../model");
var ko = require("knockout");
var LookupValuesSelectBoxOptions = (function (_super) {
    __extends(LookupValuesSelectBoxOptions, _super);
    function LookupValuesSelectBoxOptions(_parameter, _dataSourceBrowser, container) {
        var _this = _super.call(this) || this;
        _this._parameter = _parameter;
        _this._dataSourceBrowser = _dataSourceBrowser;
        _this._items = ko.observableArray();
        _this.disabled = undefined;
        _this.searchEnabled = true;
        _this.valueExpr = "Value";
        _this.searchExpr = ["Value", "DisplayText"];
        _this.displayExpr = "DisplayText";
        _this.noDataText = _default_1.getLocalizationById('DashboardStringId.FilterElementNoDataToDisplay');
        _this.placeholder = _default_1.getLocalizationById('DashboardStringId.ParametersSelectorText');
        _this.value = ko.observableArray();
        _this.showClearButton = true;
        _this.multiline = false;
        _this.showSelectionControls = true;
        _this.onDisposing = function (e) { return _this.dispose(); };
        _this.disabled = _parameter.selectAllValues;
        _this.toDispose(ko.computed(function () {
            if (!!_parameter.staticListLookUpSettings()) {
                _this._items(_parameter.staticListLookUpSettings().values().filter(function (lookUpValue) { return !!lookUpValue.value(); }).map(function (lookUpValue) { return { Value: lookUpValue.value(), DisplayText: lookUpValue.value() }; }));
            }
            else if (!!_parameter.dynamicListLookUpSettings()) {
                _this.parameterValues = _this._dataSourceBrowser.getParameterValues(_parameter.type(), _parameter.dynamicListLookUpSettings());
                if (_this.parameterValues().length === 0) {
                    _this.parameterValues.subscribe(function (newValues) { return _this._items(newValues); });
                }
                else {
                    _this._items(_this.parameterValues());
                }
            }
        }));
        _this.toDispose(_this.dataSource = ko.computed(function () {
            return new data_source_1.default({
                store: new dx_analytics_core_1.default.Analytics.Internal.SortedArrayStore(_this._items(), "displayValue"),
                paginate: true,
                pageSize: 100
            });
        }));
        _this.dropDownOptions = {
            container: container
        };
        return _this;
    }
    return LookupValuesSelectBoxOptions;
}(model_1.DisposableObject));
exports.LookupValuesSelectBoxOptions = LookupValuesSelectBoxOptions;
var LookupDefaultValuesSelectBoxOptions = (function (_super) {
    __extends(LookupDefaultValuesSelectBoxOptions, _super);
    function LookupDefaultValuesSelectBoxOptions(_parameter, _dataSourceBrowser, container) {
        var _this = _super.call(this, _parameter, _dataSourceBrowser, container) || this;
        _this.onValueChanged = function (e) {
            _this._parameter.defaultValues((e.value || []).map(function (val) {
                var lookUpValue = new look_up_value_1.LookUpValue();
                lookUpValue.value(val);
                return lookUpValue;
            }));
        };
        _this.value(_parameter._valuesOfDefaultValues.peek());
        return _this;
    }
    return LookupDefaultValuesSelectBoxOptions;
}(LookupValuesSelectBoxOptions));
exports.LookupDefaultValuesSelectBoxOptions = LookupDefaultValuesSelectBoxOptions;
var LookupDefaultValueSelectBoxOptions = (function (_super) {
    __extends(LookupDefaultValueSelectBoxOptions, _super);
    function LookupDefaultValueSelectBoxOptions(_parameter, _dataSourceBrowser, container) {
        var _this = _super.call(this, _parameter, _dataSourceBrowser, container) || this;
        _this.onValueChanged = function (e) {
            _this._parameter.defaultValue(e.value);
        };
        _this.value(_parameter.defaultValue.peek());
        return _this;
    }
    return LookupDefaultValueSelectBoxOptions;
}(LookupValuesSelectBoxOptions));
exports.LookupDefaultValueSelectBoxOptions = LookupDefaultValueSelectBoxOptions;
